sql/src/main/java/org/apache/druid/sql/calcite/rel/QueryMaker.java

```diff
@@ -126,7 +126,7 @@ public ObjectMapper getJsonMapper()
     return DataSourceAnalysis.forDataSource(query.getDataSource())
                              .getBaseQuerySegmentSpec()
                              .map(QuerySegmentSpec::getIntervals)
-                             .orElse(query.getIntervals());
+                             .orElseGet(query::getIntervals);
```

Comment:
Probably worth commenting why it should be lazy. Same for other `orElseGet()`.

Reply:
I'm not sure a comment is necessary... I think we should probably always use `orElseGet` if the 'else' isn't a primitive because `Optional` must eagerly evaluate whatever goes in there. This should maybe be done with some sort of static analysis, but I couldn't figure it out in like the 2 minutes I devoted to it so far.

Reply:
btw, I made the changes here after I made the mistake of not using `orElseGet` in `ReferenceCountingSegmentQueryRunner` and funny side-effects were happening causing integration tests to fail, so I looked for other usages of `orElse` and switched the ones that were not returning a primitive or static value to use `orElseGet`, of which there were only these 2 I think.